code/Weapons/Gadget/Components/AnvilPhysicsGadgetComponent.cs

```diff
@@ -0,0 +1,92 @@
+﻿namespace Grubs;
+
+[Prefab]
+public partial class AnvilPhysicsGadgetComponent : GadgetComponent
+{
+	[Prefab, Net]
+	public bool LockXAxis { get; set; } = true;
+
+	[Prefab, Net]
+	public bool AffectedByWind { get; set; } = false;
+
+	[Prefab, ResourceType( "sound" )]
+	public string CollisionSound { get; set; }
+
+	[Prefab, Net, Predicted]
+	public int Bounces { get; set; } = 0;
+
+	[Prefab, Net]
+	public float BounceForce { get; set; } = 100.0f;
+
+	private bool _isGrounded;
+	private bool _wasGrounded;
+
+	public override bool IsResolved()
+	{
+		return Bounces == 0 && Gadget.Velocity.IsNearlyZero( 2.5f );
+	}
+
+	public override void OnUse( Weapon weapon, int charge )
+	{
+		if ( GrubsConfig.WorldTerrainType is GrubsConfig.TerrainType.Texture )
+			Gadget.Position = Grub.Player.MousePosition.WithZ( GrubsGame.Instance.Terrain.WorldTextureHeight + 192f );
+		else
+			Gadget.Position = Grub.Player.MousePosition.WithZ( GrubsConfig.TerrainHeight + 192f );
+	}
+
+	public override void Simulate( IClient client )
+	{
+		// Apply gravity.
+		Gadget.Velocity -= new Vector3( 0, 0, 400 ) * Time.Delta;
+
+		var helper = new MoveHelper( Gadget.Position, Gadget.Velocity );
+		helper.Trace = helper.Trace
+			.Size( Gadget.CollisionBounds )
+			.WithAnyTags( Tag.Player, Tag.Solid )
+			.WithoutTags( Tag.Dead )
+			.UseHitboxes( true );
+
+		helper.TryMove( Time.Delta );
+		Gadget.Position = helper.Position;
+		Gadget.Velocity = helper.Velocity;
+		if ( LockXAxis )
+			Gadget.Velocity = Gadget.Velocity.WithX( 0 );
+
+		var trCollision = Trace.Ray( Gadget.Position, Gadget.Position )
+				.Size( Gadget.CollisionBounds * 1.1f ) // Slightly increase to make sure it collides.
+				.Ignore( Gadget )
+				.WithAnyTags( Tag.Player, Tag.Solid )
+				.WithoutTags( Tag.Shard )
+				.UseHitboxes( true )
+				.Run();
+
+		_isGrounded = trCollision.Hit;
+
+		if ( _wasGrounded != _isGrounded )
+		{
+			_wasGrounded = _isGrounded;
+			if ( _isGrounded )
+				OnCollision();
+		}
+
+		if ( GrubsConfig.WindEnabled && AffectedByWind )
+			Gadget.Velocity += new Vector3( GamemodeSystem.Instance.ActiveWindForce ).WithY( 0 );
+	}
+
+	private void OnCollision()
+	{
+		Gadget.PlaySound( CollisionSound );
```

Comment:
Do you think we should include a collision sound?

Reply:
I added this in case we ever wanted a weapon such as say, an anvil, that just falls and hits the ground instead of exploding. For Concrete Garry, however, I don't see the need for a collision sound.